### ISSUE TITLE
Resize does not average properly

### ISSUE THREAD

**Issue by drahnr** [image-rs/image#1468]
The following unit test used to work:

https://github.com/spearow/juice/blob/master/greenglas/tests/image_spec.rs#L109-L118

The image was a `2x2` RGB Pixel with 3 white `(255,255,255)` and one black `(0,0,0)` pixel.

Resizing this to an image of size `1x1` then...

## Expected

`image = 0.23.11` resolved this as `(191, 191, 191)` which is what I'd expect when averaging all of them.

## Actual behaviour

The result as of `image = 0.23.14` is `(192, 192, 192)` which unexpected.

## Reproduction steps

https://github.com/spearow/juice/blob/master/greenglas/tests/image_spec.rs#L109-L118

**Comment by drahnr** [comment]
Change was introduced in `0.23.12`

**Comment by fintelia** [comment]
I believe this is caused by doing the down-sampling in two steps. We first resize to a 1x2 image with contents [(255, 255, 255), (128, 128, 128)], and then to a 1x1 image from there. The double rounding causes an off-by-one error.

**Comment by drahnr** [comment]
The mentioned impl detail changes the behavior of public API without noting. So imho this is a bug. The simple solution would be to add accuracy to the result, but given the current intermediate `ImageBuffer` type prevents that easy approach.

if there is any input before I take a stab at this, I'd be happy to get some input.

**Comment by fintelia** [comment]
Agreed that this is a bug. I think the most straightforward direction might be to get rid of the two step down-sampling approach entirely, so that we don't have to worry about storing intermediate results.

**Comment by 197g** [comment]
We're still sampling twice, but at least no longer exhibiting this bug since there is not intermediate rounding. All the internals of the operation stays in RGBF32 space.

```
extern crate image; // 0.25.6

fn main() {
    let n = image::RgbImage::from_vec(
        2,
        2,
        vec![255, 255, 255, 255, 255, 255, 0, 0, 0, 255, 255, 255],
    )
    .unwrap();
    let a = image::imageops::resize(&n, 1, 1, image::imageops::FilterType::Lanczos3);
    assert_eq!(a[(0, 0)], image::Rgb([191, 191, 191]));
}
```

Feel free to reopen if there's still a concern about resizing not covered by other issues.
